core-tests/shared/src/test/scala/zio/RuntimeSpec.scala

```diff
@@ -0,0 +1,44 @@
+package zio
+
+import zio.test.{TestAspect, assertCompletes}
+
+import java.util.concurrent.atomic.AtomicBoolean
+
+object RuntimeSpec extends ZIOBaseSpec {
+  private implicit val unsafe: Unsafe = Unsafe.unsafe
+
+  private val rt = Runtime.default.unsafe
+
+  def spec = suite("RuntimeSpec")(
+    test("unsafe.fork on synchronous effects") {
+      ZIO.succeed {
+        val (f1, f2) = mkEffects()
+        rt.fork(f1)
+        rt.fork(f2)
+        assertCompletes
+      }
+    },
+    test("unsafe.runToFuture on synchronous effects") {
+      ZIO.suspendSucceed {
+        val (f1, f2) = mkEffects()
+        val fut1     = rt.runToFuture(f1)
+        val fut2     = rt.runToFuture(f2)
+        ZIO.fromFuture(_ => fut1) *> ZIO.fromFuture(_ => fut2) *> assertCompletes
+      }
+    }
+  ) @@ TestAspect.timeout(5.seconds)
+
+  private def mkEffects() = {
+    val loop = new AtomicBoolean(true)
+
+    val f1 = ZIO.succeed {
```

Comment:
Better to do something like `(ZIO.succeed(loop.get()) <* ZIO.yieldNow).retryUntil(_ == true)` so it works on JS too and will be more efficient.

Reply:
Then you can do `@@ jvmOnly` or equivalent to run it only on the JVM.

Reply:
Agh I tried that but turns out it fails to compile with Scala native due to `Thread.sleep`. I moved the test suite under `/jvm` instead